MetaLamp/lending-pool/test/Spec/User.hs

```diff
@@ -0,0 +1,140 @@
+lenderWallet = Wallet 2
+
+borrowerWallet :: Wallet
+borrowerWallet = Wallet 3
+
+userWallets :: [Wallet]
+userWallets = [lenderWallet, borrowerWallet]
+
+initialFunds :: Value
+initialFunds = lovelaceValueOf 1000000 <> mconcat ((`assetClassValue` 1000) <$> testAssets)
+
+type UserHandle = TestUtils.ContractHandle Text Aave.UserContractState Aave.AaveUserSchema Void
+
+startTrace :: Trace.EmulatorTrace (Map.Map Wallet UserHandle)
+startTrace = do
+    _ <- Trace.activateContractWallet ownerWallet startContract
+    _ <- Trace.waitNSlots 5
+    _ <- Trace.activateContractWallet ownerWallet $ initContract userWallets testAssets
+    _ <- Trace.waitNSlots 5
+    fmap Map.fromList $ forM userWallets $ \wallet -> do
+        handle <- Trace.activateContractWallet wallet userContract
+        pure (wallet, handle)
+
+getPubKey :: UserHandle -> Trace.EmulatorTrace Ledger.PubKeyHash
+getPubKey userHandle = do
+    _ <- Trace.callEndpoint @"ownPubKey" userHandle ()
+    _ <- Trace.waitNSlots 1
+    TestUtils.getState (^? Aave._GetPubKey) userHandle
+
+deposit :: UserHandle -> AssetClass -> Integer -> Trace.EmulatorTrace ()
+deposit userHandle asset amount = do
+    pkh <- getPubKey userHandle
+    Trace.callEndpoint @"deposit" userHandle $ Aave.DepositParams asset pkh amount
+    _ <- Trace.waitNSlots 3
+    pure ()
+
+withdraw :: UserHandle -> AssetClass -> Integer -> Trace.EmulatorTrace ()
+withdraw userHandle asset amount = do
+    pkh <- getPubKey userHandle
+    Trace.callEndpoint @"withdraw" userHandle $ Aave.WithdrawParams asset pkh amount
+    _ <- Trace.waitNSlots 3
+    pure ()
+
+initialReserves :: AssocMap.Map AssetClass Aave.Reserve
+initialReserves = AssocMap.fromList (fmap (\params -> (Aave.cpAsset params, Aave.createReserve Mock.aave params)) startParams)
+
+modifyAt :: PlutusTx.Eq k => (v -> v) -> k -> AssocMap.Map k v -> AssocMap.Map k v
+modifyAt f k m = maybe m (\v -> AssocMap.insert k (f v) m) (AssocMap.lookup k m)
+
+modifyAmount :: (Integer -> Integer) -> AssetClass -> AssocMap.Map AssetClass Aave.Reserve -> AssocMap.Map AssetClass Aave.Reserve
+modifyAmount f = modifyAt (\r -> r { Aave.rAmount = f . Aave.rAmount $ r })
+
+reservesChange :: AssocMap.Map AssetClass Aave.Reserve -> TracePredicate
+reservesChange reserves = TestUtils.datumsAtAddress Mock.aaveAddress (TestUtils.one check)
+    where
+        check (Aave.ReservesDatum _ reserves') = reserves' == reserves
+        check _                                = False
+
+tests :: TestTree
+tests = testGroup "deposit" [
+    checkPredicate
+        "Successful deposit"
+        (walletFundsChange
+            lenderWallet
+            (initialFunds <>
+            assetClassValue mogus (negate 100) <> assetClassValue amogus 100)
+        .&&. reservesChange (modifyAmount (+100) mogus initialReserves)
+        )
+        $ do
+            handles <- startTrace
+            deposit (handles Map.! lenderWallet) mogus 100,
+    checkPredicate
+        "Successful withdraw"
```

Comment:
Let's create a more descriptive header, e.g.
        "Should successfully withdraw after depositing a bigger amount"